zipline/algorithm.py

```diff
@@ -907,6 +940,7 @@ def _calculate_order_value_amount(self, asset, value):
         return value / (last_price * value_multiplier)
 
     @api_method
+    @check_asset_lifetime
     def order(self, asset, amount,
```

Comment:
agreed.  OK with me making it all `asset`?  hopefully that doesn't cause too many legacy breakages.